api_app/analyzers_manager/file_analyzers/intezer_scan.py

```diff
@@ -33,59 +35,107 @@ def _get_access_token(api_key):
     response_json = response.json()
     token = response_json.get("result", "")
     os.environ["INTEZER_TOKEN"] = token
-    os.environ["INTEZER_TOKEN_DATE"] = get_now_date_only()
+    os.environ["INTEZER_TOKEN_DATE"] = get_now_str()
     return token
 
 
+def get_access_token(api_key):
+    intezer_token = os.environ.get("INTEZER_TOKEN", "")
+    intezer_token_date = os.environ.get("INTEZER_TOKEN_DATE", None)
+    intezer_token_datetime_object = None
+    if intezer_token_date:
+        intezer_token_datetime_object = datetime.strptime(
+            intezer_token_date, "%Y-%m-%d %H:%M:%S"
+        )
+    now = get_now()
+    if (
+        not intezer_token
+        or not intezer_token_datetime_object
+        or (intezer_token_datetime_object < now - timedelta(hours=9))
+    ):
+        intezer_token = _retrieve_access_token_from_intezer(api_key)
+        if not intezer_token:
+            raise AnalyzerRunException("token extraction failed")
+
+    return intezer_token
```

Comment:
Why is the Intezer API so complicated ? I really didn't think about this before but the we are writing to the `os.environ` which can be super buggy when used with celery. 

Maybe instead of all this logic we should use their Python SDK (https://github.com/intezer/analyze-python-sdk) ?

It provides a very simple interface:

```python3
from intezer_sdk import consts

index = Index(file_path=<file_path>, 
              index_as=consts.IndexType.MALICIOUS, 
              family_name=<family_name>)
index.send(wait=True)
index_id = index.index_id
```

Reply:
yeah, the env variables are a concern. Considering that my main goal was to add new functionalities, I added even other complexity. Even if I would always try to avoid the usage of external packages if unecessary, if that sdk works correctly it would really make this analyzer simpler and without problems. I'll test it and update this PR. Thank you for the suggestions

Reply:
I have no time to end this before my vacation. Either you just get the fix for Quark and I'll finish this later or someone else should finish this.

Reply:
I am in contact with the Intezer team to make official our integration so I just need to end this until the end of the month. No particular hurry otherwise

Reply:
No worries, I'll finish the work in this PR.